apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/PatientClaimsEobTaskTransformerV2Test.java

```diff
@@ -0,0 +1,580 @@
+    when(mockCoding.getCode()).thenReturn("CARRIER");
+  }
+
+  /** sets up claim entities. */
+  private void setupEntities() {
+    List<Object> parsedRecords =
+        ServerTestUtils.parseData(Arrays.asList(StaticRifResourceGroup.SAMPLE_A.getResources()));
+
+    testCarrierClaim =
+        parsedRecords.stream()
+            .filter(r -> r instanceof CarrierClaim)
+            .map(CarrierClaim.class::cast)
+            .findFirst()
+            .get();
+
+    testDmeClaim =
+        parsedRecords.stream()
+            .filter(r -> r instanceof DMEClaim)
+            .map(DMEClaim.class::cast)
+            .findFirst()
+            .get();
+
+    testHhaClaim =
+        parsedRecords.stream()
+            .filter(r -> r instanceof HHAClaim)
+            .map(HHAClaim.class::cast)
+            .findFirst()
+            .get();
+
+    testHospiceClaim =
+        parsedRecords.stream()
+            .filter(r -> r instanceof HospiceClaim)
+            .map(HospiceClaim.class::cast)
+            .findFirst()
+            .get();
+
+    testInpatientClaim =
+        parsedRecords.stream()
+            .filter(r -> r instanceof InpatientClaim)
+            .map(InpatientClaim.class::cast)
+            .findFirst()
+            .get();
+
+    testOutpatientClaim =
+        parsedRecords.stream()
+            .filter(r -> r instanceof OutpatientClaim)
+            .map(OutpatientClaim.class::cast)
+            .findFirst()
+            .get();
+    testPdeClaim =
+        parsedRecords.stream()
+            .filter(r -> r instanceof PartDEvent)
+            .map(PartDEvent.class::cast)
+            .findFirst()
+            .get();
+
+    testSnfClaim =
+        parsedRecords.stream()
+            .filter(r -> r instanceof SNFClaim)
+            .map(SNFClaim.class::cast)
+            .findFirst()
+            .get();
+  }
+
+  /**
+   * Verify that the {@link PatientClaimsEobTaskTransformerV2} sucessfully transforms a {@link
+   * CarrierClaim} entity into a {@link ExplanationOfBenefit} FHIR resource and performs no SAMHSA
+   * EOB filtering.
+   */
+  @Test
+  void testTaskTransformerUsingCarrierClaimNoSamhsaFilter() throws IOException {
+    CriteriaQuery<CarrierClaim> clmMockCriteria = mock(CriteriaQuery.class);
+    Root<CarrierClaim> clmRoot = mock(Root.class);
+    setupClaimEntity(mockEntityManager, ClaimTypeV2.CARRIER, clmMockCriteria, clmRoot);
+
+    ClaimTransformerInterfaceV2 claimTransformer =
+        new CarrierClaimTransformerV2(metricRegistry, mockDrugDisplayLookup, mockNpiOrgLookup);
+    PatientClaimsEobTaskTransformerV2 taskTransformer =
+        new PatientClaimsEobTaskTransformerV2(
+            metricRegistry, mockSamhsaMatcher, mockDrugDisplayLookup, mockNpiOrgLookup);
+
+    taskTransformer.setIncludeTaxNumbers(true);
+    taskTransformer.setupTaskParams(
+        claimTransformer, ClaimTypeV2.CARRIER, 1234L, Optional.empty(), Optional.empty(), false);
+    taskTransformer.setEntityManager(mockEntityManager);
+
```

Comment:
It'd be nice to add a verification the samhsa matcher was not called here (similar to the verify that determined no interactions occurred on the old tests)

Reply:
done.